### PR TITLE
labeler: keep compatibility when meeting duplicate rules.

### LINKED ISSUE #7747: ticdc ci report `etcdserver: duplicate key given in txn request`
## Flaky Test

after #7738, we handle `/config/region-label/rules` with txn, if there are duplicate keys in the request, it will report this kind of error.

so we could need to keep compatibility about deleting and setting rules.

But I am not sure whether we need to do the same compatibility thing about placement rules.

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #7747

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/schedule/labeler/labeler.go [hunk 1]
```diff
@@ -272,9 +272,20 @@ func (l *RegionLabeler) Patch(patch LabelRulePatch) error {
 		}
 	}
 
+	// Handling duplicate entries in DeleteRules and SetRules
+	// Note: We maintain compatibility with the previous behavior, which is to process DeleteRules before SetRules
+	// If there are duplicate rules, we will prioritize SetRules
+	deleteRules := make(map[string]struct{})
+	for _, ruleID := range patch.DeleteRules {
```
Comment (JmPotato) [inline pkg/schedule/labeler/labeler.go:279]: Iterate and build the set rules first could eliminate one traversal. We just need to skip the rules in set later while building the delete ruels.

**Review by rleungx — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Pull request merged**